model/Metadata/Service/AdvancedSearchSettingsService.php

```diff
@@ -196,4 +203,21 @@ public function getSettingsByClassMetadataSearchRequest(
 
         return new SearchSettings($out);
     }
+
+    private function getPropertiesToHide(): array
+    {
+        $propertiesToHide = self::OMIT_PROPERTIES;
+
+        foreach ($this->featureBasedPropertiesService->getAllProperties() as $featureFlag => $properties) {
```

Comment:
- [ ] Would not be better to make this service hold this logic instead of repeat it in the places we need? Example:

There is a design pattern `Filter` used for that https://www.tutorialspoint.com/design_pattern/filter_pattern.htm

```
$this->featureBasedPropertiesFilter->filter(array $properties): array
```